sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -460,14 +461,22 @@ public void accept(WindowedValue input) throws Exception {
       // Extract out relevant TimerFamilySpec information in preparation for execution.
       for (Map.Entry<String, TimerFamilySpec> entry :
           parDoPayload.getTimerFamilySpecsMap().entrySet()) {
-        String timerFamilyId = entry.getKey();
-        TimeDomain timeDomain =
-            DoFnSignatures.getTimerSpecOrThrow(
-                    doFnSignature.timerDeclarations().get(timerFamilyId), doFn)
-                .getTimeDomain();
+        String timerIdOrTimerFamilyId = entry.getKey();
+        TimeDomain timeDomain;
+        if (timerIdOrTimerFamilyId.startsWith(TimerFamilyDeclaration.PREFIX)) {
+          timeDomain =
```

Comment:
The `TTimerFamilySpec` should have `time_domain ` field. Maybe we could  do something similar to https://github.com/apache/beam/blob/1de50c348706ed25af2bab9c9477d7d4f36ef8bf/runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ParDoTranslation.java#L657-L668

Reply:
done.